### ISSUE TITLE
Failed to download 20.3.8.26

### ISSUE THREAD

**Issue by komitake** [kerl/kerl#378]
```
❯ kerl build 20.3.8.26
Downloading 20.3.8.26 to /Users/USERNAME/.kerl/archives...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
curl: (22) The requested URL returned error: 404 
```

It seems working well if I revert the commit 12c31648df1f3a7c712fdc1e9126e33804a9947a

**Comment by paulo-ferraz-oliveira** [comment]
The file it tries to download is https://github.com/erlang/otp/releases/download/OTP-20.3.8.26/otp_src_20.3.8.26.tar.gz.

I think it's the `--fail` that's actually making it exit with an error code.

if you take that out the output is "Not Found", which we could match on, but it may be kind of brittle. Maybe using the http_code on write-out?

**Comment by paulo-ferraz-oliveira** [comment]
eg. 
```
> HTTP=$(curl --silent --output /dev/null --location https://github.com/erlang/otp/releases/download/OTP-20.3.8.26/otp_src_20.3.8.26.tar.gz --write-out '%{http_code}')
> echo $HTTP
404
```

No stdout output, no error, but an HTTP error code (which might change, but it's more robust then the body).

**Comment by paulo-ferraz-oliveira** [comment]
@garazdawi, thoughts?

**Comment by jadeallenx** [comment]
I have previously used `%{http_code}` in some other bash automation and I think that's the easiest thing to use with shell integrations, so let's go that way.

**Comment by garazdawi** [comment]
Sigh, sorry for the mess up. The fix in #379 is correct.

As punishment, I will try to add some tests that makes sure that this does not break again in the future.

Thanks for reporting @komitake and also, just so that you know, Erlang/OTP 20 stopped being supported and thus receiving security patches 1½ years ago. So you may want to think about upgrading to at least Erlang/OTP 22.
